lib/core/src/display/shape/text/text_field/keyboard.rs

```diff
@@ -0,0 +1,197 @@
+//! A FRP definitions for keyboard event handling, with biding this FRP graph to js events.
+
+use crate::prelude::*;
+
+use crate::display::shape::text::text_field::cursor::Step;
+use crate::display::shape::text::text_field::TextFieldData;
+use crate::system::web::text_input::KeyboardBinding;
+
+use enso_frp::*;
+use web_sys::KeyboardEvent;
+
+
+
+// ====================
+// === TextFieldFrp ===
+// ====================
+
+/// This structure contains all nodes in FRP graph handling keyboards events of one TextField
+/// component.
+///
+/// The most of TextField actions are covered by providing actions to KeyboardActions for specific
+/// key masks. However, there are special actions which must be done in a lower level:
+///  * *clipboard operations* - they are performed by reading text input js events directly from
+///    text area component. See `system::web::text_input` crate.
+///  * *text input operations* - here we want to handle all the keyboard mapping set by user, so
+///    we connect this action directly to `key_press` node from `keyboard`.
+#[derive(Debug)]
+pub struct TextFieldFrp {
+    /// A "keyboard" part of graph derived from frp crate.
+    keyboard: Keyboard,
+    /// Keyboard actions. Here we define shortcuts for all actions except letters input, copying
+    /// and pasting.
+    actions: KeyboardActions,
+    /// A node producing event once cut operation was requested.
```

Comment:
After thinking for a while about it, I dont like these long docs. `FRP Node` is an internal term. For users these are just actions. Why dont we write instead `Produces event once cut operation was requested.` ?

Reply:
But the care about our docs to be nice (that's why we put `.` everywhere), and sentences without subject are not nice to me.